uniffi_bindgen/src/bindings/python/gen_python/mod.rs

```diff
@@ -108,6 +111,56 @@ static KEYWORDS: Lazy<HashSet<String>> = Lazy::new(|| {
     HashSet::from_iter(kwlist.into_iter().map(|s| s.to_string()))
 });
 
+/// The main public interface for this module, representing the complete details of an interface exposed
+/// by a rust component and the details of consuming it via an extern-C FFI layer.
+#[derive(Debug, Default)]
+pub struct PythonComponentInterface {
+    /// All of the types used in the interface.
+    // We can't checksum `self.types`, but its contents are implied by the other fields
+    // anyway, so it's safe to ignore it.
+    pub(super) types: TypeUniverse,
+    /// The high-level API provided by the component.
+    enums: BTreeMap<String, Enum>,
+    records: BTreeMap<String, Record>,
+    functions: Vec<Function>,
+    objects: Vec<Object>,
+    callback_interfaces: Vec<CallbackInterface>,
+    // Type names which were seen used as an error.
+    errors: HashSet<String>,
+    // Types which were seen used as callback interface error.
+    callback_interface_throws_types: BTreeSet<Type>,
+}
+
+impl_language_component_interface!(PythonComponentInterface);
+
+impl PythonComponentInterface {
+    /// Get a Record definition by name
+    pub fn get_record_definition_filter<'a>(&self, name: &str) -> askama::Result<&'a Record> {
+        self.get_record_definition(name)
+            .ok_or_else(|| lookup_error!("record {name} not found"))
+    }
+
+    /// Get an Object definition by name
+    pub fn get_object_definition_filter<'a>(&self, name: &str) -> askama::Result<&'a Object> {
+        self.get_object_definition(name)
+            .ok_or_else(|| lookup_error!("object {name} not found"))
+    }
+}
+impl From<ComponentInterface> for PythonComponentInterface {
```

Comment:
This might be where we apply naming conventions etc. So we just iterate once over all the structures and apply the changes.